verification/verification_test.go

```diff
@@ -198,3 +198,59 @@ func testInitContext(d TestDPEInstance, t *testing.T) {
 		}
 	}
 }
+
+func TestCertifyKey(t *testing.T) {
+	simulators := []TestDPEInstance{
+		// No extra options besides AutoInit.
+		&DpeSimulator{exe_path: *sim_exe, supports: Support{AutoInit: true}},
+		// Supports AutoInit and simulation contexts.
+		&DpeSimulator{exe_path: *sim_exe, supports: Support{AutoInit: true, Simulation: true}},
+	}
+
+	for _, s := range simulators {
+		s.SetLocality(DPE_SIMULATOR_AUTO_INIT_LOCALITY)
+		testCertifyKey(s, t)
+	}
+}
+
+func testCertifyKey(d TestDPEInstance, t *testing.T) {
+	if d.HasPowerControl() {
+		err := d.PowerOn()
+		if err != nil {
+			log.Fatal(err)
+		}
+		defer d.PowerOff()
+	}
+	client, err := NewClient256(d)
+	if err != nil {
+		t.Fatalf("Could not initialize client: %v", err)
+	}
+
+	certifyKeyReq := CertifyKeyReq[SHA256Digest]{
+		ContextHandle: [16]byte{0},
+		Flags:         0,
+		Label:         [32]byte{0},
+	}
+	certifyKeyResp, err := client.CertifyKey(&certifyKeyReq)
+	if err != nil {
+		t.Fatalf("Could not certify key: %v", err)
+	}
+	t.Logf("Certificate: %x", certifyKeyResp.Certificate)
```

Comment:
Should we do any initial checks to see if the certificate that came back can be parsed. Or would that be better in a follow on PR?

Reply:
That's a good idea, let me try something basic here. I expect this test will grow a bit in the future (for example, to compare the keys involved in CertifyKey and Sign)

Reply:
With what you found it sounds like it would be better in a follow-on PR.

Reply:
Sounds good to me. (For posterity, when I tried to implement this I found that the serial numbers produced by the DPE are not printable strings)